### PR TITLE
Configurable metrics

### PR DESCRIPTION
**Pull request by djdameln**
# Description

- This PR makes metrics configurable from the config.yaml. The config file is currently only updated for PADIM. Will update the other ones after feedback on this approach.
- Fixes https://github.com/openvinotoolkit/anomalib/issues/226, https://github.com/openvinotoolkit/anomalib/issues/199

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/models/components/base/anomaly_module.py [hunk 1]
```diff
@@ -58,12 +57,9 @@ def __init__(self, params: Union[DictConfig, ListConfig]):
         self.model: nn.Module
 
         # metrics
-        image_auroc = AUROC(num_classes=1, pos_label=1, compute_on_step=False)
-        image_f1 = F1(num_classes=1, compute_on_step=False, threshold=self.hparams.model.threshold.image_default)
-        pixel_auroc = AUROC(num_classes=1, pos_label=1, compute_on_step=False)
-        pixel_f1 = F1(num_classes=1, compute_on_step=False, threshold=self.hparams.model.threshold.pixel_default)
-        self.image_metrics = MetricCollection([image_auroc, image_f1], prefix="image_").cpu()
-        self.pixel_metrics = MetricCollection([pixel_auroc, pixel_f1], prefix="pixel_").cpu()
+        self.image_metrics, self.pixel_metrics = get_metrics(self.hparams)
```
Comment (ashwinvaidya17) [inline anomalib/models/components/base/anomaly_module.py:60]: Is there something planned for classification models which do not have pixel metrics? When I removed the pixel metric key from the config file it threw error for padim

Comment (djdameln) [reply anomalib/models/components/base/anomaly_module.py:60]: This should work:
```
metrics:
  image:
    - F1
    - AUROC
  pixel: []
```

Or this:
  ```
  metrics:
  image:
    - F1
    - AUROC
  pixel: None

Comment (ashwinvaidya17) [reply anomalib/models/components/base/anomaly_module.py:60]: Works now 🙂

File: anomalib/utils/metrics/__init__.py [hunk 2]
```diff
@@ -1,8 +1,61 @@
...
 from .anomaly_score_distribution import AnomalyScoreDistribution
 from .auroc import AUROC
+from .collection import AnomalibMetricCollection
 from .min_max import MinMax
 from .optimal_f1 import OptimalF1
 
 __all__ = ["AUROC", "OptimalF1", "AdaptiveThreshold", "AnomalyScoreDistribution", "MinMax"]
+
+
+def get_metrics(config: Union[ListConfig, DictConfig]) -> Tuple[AnomalibMetricCollection, AnomalibMetricCollection]:
```
Comment (samet-akcay) [inline anomalib/utils/metrics/__init__.py:19]: We might need to modify this for LightningCLI

File: anomalib/models/components/base/anomaly_module.py (hunk 1 above)
Comment (samet-akcay) [reply anomalib/models/components/base/anomaly_module.py:60]: It would be good to handle a case such as following:
```yaml
metrics:
  image:
    - F1
    - AUROC
```
so `pixel: []` would be assigned under the hood.

Comment (samet-akcay) [reply anomalib/models/components/base/anomaly_module.py:60]: It would be good to handle a case such as following:
```yaml
metrics:
  image:
    - F1
    - AUROC
```
so `pixel: []` would be assigned under the hood.

Comment (samet-akcay) [reply anomalib/models/components/base/anomaly_module.py:60]: Or what would be the behaviour for the following:
```yaml
metrics:
 - F1
 - AUROC
```

Comment (samet-akcay) [reply anomalib/models/components/base/anomaly_module.py:60]: Or what would be the behaviour for the following:
```yaml
metrics:
 - F1
 - AUROC
```

**Comment by alexriedel1** [comment]
Works like a charm!

File: anomalib/models/components/base/anomaly_module.py (hunk 1 above)
Comment (samet-akcay) [reply anomalib/models/components/base/anomaly_module.py:60]: Not sure why my comments disappeared from here, but ideally 
```yaml
metrics:
  image:
    - F1
    - AUROC
```

should also work fine.

**Comment by ashwinvaidya17** [comment]
Might be out of the scope of this PR but we can have a look at this PR https://github.com/PyTorchLightning/metrics/pull/709 which updates the base metrics only once. Maybe we can find a way to automatically group metrics.

**Comment by ashwinvaidya17** [comment]
Can you fix torchmetrics version in this PR as well `>=0.8.0`

**Review by samet-akcay — APPROVED** [review]
